networking/p2p/src/main/java/tech/pegasys/teku/networking/p2p/discovery/discv5/NodeRecordConverter.java

```diff
@@ -29,9 +36,19 @@
 
   private static DiscoveryPeer socketAddressToDiscoveryPeer(
       final NodeRecord nodeRecord, final InetSocketAddress address) {
+
+    Optional<EnrForkId> enrForkId =
+        Optional.ofNullable((Bytes) nodeRecord.get(ETH2_ENR_FIELD))
+            .map(enrField -> SimpleOffsetSerializer.deserialize(enrField, EnrForkId.class));
+
+    Bitvector persistentSubnets =
+        Optional.ofNullable((Bytes) nodeRecord.get(ATTESTATION_SUBNET_ENR_FIELD))
+            .map(
+                attestionSubnetsField ->
+                    Bitvector.fromBytes(attestionSubnetsField, ATTESTATION_SUBNET_COUNT))
```

Comment:
This may throw an `IllegalArgumentException` if the bytes are not the right length.

Reply:
I needed to build on this so have added error handling for both ENR fields and merged. :)